Canary live stream

snjoetw opened the pull request:
## Description:
Added support for Canary live stream. 

## Breaking Change
Originally Canary camera is added per location and only displays an image that was captured due to motion. Now it is per device (each location can have multiple devices) with live stream support.

## Checklist:
  - [x] The code change is tested and works locally.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

snjoetw committed:
Added support for Canary live stream view

snjoetw committed:
Updated requirements

snjoetw committed:
Merge branch 'dev' of https://github.com/home-assistant/home-assistant into canary_live_stream

snjoetw committed:
- Fixed lint error

MartinHjelmare commented on homeassistant/components/camera/canary.py:
```diff
@@ -4,19 +4,30 @@
-ATTR_MOTION_END_TIME = "motion_end_time"
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_FFMPEG_ARGUMENTS): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL, default=timedelta(seconds=90)):
```
Remove this and define `SCAN_INTERVAL` with the default value at the module level. This config option is already part of the base platform schema.

MartinHjelmare commented on homeassistant/components/camera/canary.py:
```diff
@@ -25,71 +36,82 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         super().__init__()
+
+        self._ffmpeg = hass.data[DATA_FFMPEG]
+        self._ffmpeg_arguments = config.get(CONF_FFMPEG_ARGUMENTS)
```
Why pass the whole config if only this option is needed?

MartinHjelmare commented on homeassistant/components/camera/canary.py:
```diff
@@ -25,71 +36,82 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
+
+    @property
+    def should_poll(self):
+        """Update the image periodically."""
```
This doesn't update the image but the session, right?

snjoetw replied:
Correct this is just to extend the session

MartinHjelmare commented on homeassistant/components/camera/canary.py:
```diff
@@ -25,71 +36,82 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
+    @property
+    def should_poll(self):
+        """Update the image periodically."""
+        return True
```
Why do you need polling?

snjoetw replied:
This is so that live stream session can be extended periodically

MartinHjelmare replied:
Set a throttle on the method that does that and just call it before getting the image. Ie, don't activate poll here. The frontend should take care of this.

snjoetw committed:
Merge branch 'dev' of https://github.com/home-assistant/home-assistant into canary_live_stream

snjoetw committed:
Addressed PR comment

snjoetw committed:
- Disabled polling for Canary camera as suggested in PR comment
- Live session is now renewed every time camera is retrieved and min time between session renewal is 90 seconds

MartinHjelmare reviewed (approved):
Looks great!

MartinHjelmare commented:
Please single out the paragraph in the description about the breaking change, so that it's easy to add it to the release notes. Add a header **Breaking change** before the paragraph.

The docs need updating to describe that the ffmpg component is needed, correct? Please add a link to a docs PR, then we can merge this.

snjoetw commented:
Updated description with breaking change section and submitted PR for doc changes
